test/unit/storage/LockingResourceStore.test.ts

```diff
@@ -121,12 +124,41 @@ describe('A LockingResourceStore', (): void => {
         resolve();
       });
     });
+
     representation.data.destroy(new Error('Error on the Readable :('));
+
+    await handleError;
+    expect(locker.acquire).toHaveBeenCalledTimes(1);
+    expect(locker.acquire).toHaveBeenLastCalledWith({ path: 'path' });
+    expect(source.getRepresentation).toHaveBeenCalledTimes(1);
+
+    // Both the end and the close event will be invoked.
+    expect(lock.release).toHaveBeenCalledTimes(2);
+    expect(order).toEqual([ 'acquire', 'getRepresentation', 'error', 'release', 'release' ]);
+  });
+
+  it('releases the lock on the resource when readable is destroyed.', async(): Promise<void> => {
+    const representation = await store.getRepresentation({ path: 'path' }, {});
+    const handleError = new Promise((resolve): any => {
+      representation.data.prependListener('close', (): any => {
+        order.push('close');
+        resolve();
+      });
+    });
+
+    representation.data.destroy();
+
     await handleError;
     expect(locker.acquire).toHaveBeenCalledTimes(1);
     expect(locker.acquire).toHaveBeenLastCalledWith({ path: 'path' });
     expect(source.getRepresentation).toHaveBeenCalledTimes(1);
     expect(lock.release).toHaveBeenCalledTimes(1);
-    expect(order).toEqual([ 'acquire', 'getRepresentation', 'error', 'release' ]);
+    expect(order).toEqual([ 'acquire', 'getRepresentation', 'close', 'release' ]);
+  });
+
+  it('releases the lock without error when called twice.', async(): Promise<void> => {
+    const dummyLock = await new SingleThreadedResourceLocker().acquire({ path: 'dummyIdentifier' });
+    await dummyLock.release();
+    await dummyLock.release();
```

Comment:
This test should be in `SingleThreadedResourceLocker.test.ts`, there should be no reference to a specific locker in this file. Probably should also add some `expects` showing what is expected to happen.

Reply:
There are no expects because the test checks that no errors are thrown, and the test fails when an unexpected error occurs.
However I think I could make it more clear if I change it to `await expect(dummyLock.release()).resolves.not.toThrow(Error);`?

I'll move it to `SingleThreadedResourceLocker.test.ts` as well.

Reply:
Just `await expect(dummyLock.release()).resolves.toBeUndefined()` which tests that it fully resolves.